migration/alembic/versions/d1dc98f61aee_major_changes.py

```diff
@@ -0,0 +1,142 @@
+    op.add_column('file', sa.Column('ext_src', sa.String(), nullable=True))
+    op.add_column('job', sa.Column('ext_id', sa.Integer(), nullable=True))
+    op.add_column('job', sa.Column('ext_src', sa.String(), nullable=True))
+    op.add_column('location', sa.Column('ext_id', sa.Integer(), nullable=True))
+    op.add_column('location', sa.Column('ext_src', sa.String(), nullable=True))
+    op.add_column('metric', sa.Column('ext_id', sa.Integer(), nullable=True))
+    op.add_column('metric', sa.Column('ext_src', sa.String(), nullable=True))
+    op.add_column('operation', sa.Column('reference_id', sa.Integer(), nullable=True))
+    op.add_column('operation', sa.Column('ext_id', sa.Integer(), nullable=True))
+    op.add_column('operation', sa.Column('ext_src', sa.String(), nullable=True))
+    op.create_foreign_key(None, 'operation', 'reference', ['reference_id'], ['id'])
+    op.add_column('operation_config', sa.Column('ext_id', sa.Integer(), nullable=True))
+    op.add_column('operation_config', sa.Column('ext_src', sa.String(), nullable=True))
+    op.add_column('project', sa.Column('ext_id', sa.Integer(), nullable=True))
+    op.add_column('project', sa.Column('ext_src', sa.String(), nullable=True))
+    op.drop_column('project', 'fms_id')
+    op.add_column('readset', sa.Column('state', postgresql.ENUM('VALID', 'ON_HOLD', 'INVALID', name='stateenum', create_type=False), nullable=True))
+    op.add_column('readset', sa.Column('ext_id', sa.Integer(), nullable=True))
+    op.add_column('readset', sa.Column('ext_src', sa.String(), nullable=True))
+    op.drop_column('readset', 'quality_offset')
+    op.add_column('run', sa.Column('ext_id', sa.Integer(), nullable=True))
+    op.add_column('run', sa.Column('ext_src', sa.String(), nullable=True))
+    op.drop_column('run', 'fms_id')
+    # Change sequence name Cf. https://petegraham.co.uk/rename-postgres-table-with-alembic/
+    op.execute('ALTER SEQUENCE patient_id_seq RENAME TO specimen_id_seq')
+    op.add_column('sample', sa.Column('ext_id', sa.Integer(), nullable=True))
+    op.add_column('sample', sa.Column('ext_src', sa.String(), nullable=True))
+    # Rename column patient_id to specimen_id in sample table
+    op.alter_column('sample', 'patient_id', new_column_name='specimen_id')
+    op.drop_column('sample', 'fms_id')
+    op.drop_column('specimen', 'fms_id')
+    public_to_upper = PGFunction(
+        schema="public",
+        signature="to_upper(some_text text)",
+        definition='RETURNS text as\n  $$\n    SELECT upper(some_text)\n  $$ language SQL'
+    )
+    op.create_entity(public_to_upper)
+
+    op.sync_enum_values('public', 'flagenum', ['PASS', 'WARNING', 'FAILED', 'MISSING', 'NOT_APPLICABLE'],
+                        [TableReference(table_schema='public', table_name='metric', column_name='flag')],
+                        enum_values_to_rename=[])
+    # ### end Alembic commands ###
+
+
+def downgrade() -> None:
```

Comment:
I don't see the reestablishment of fms_id in the downgrade?

Reply:
I see it lines 104 and 111 and 118.